nano/secure/blockstore.hpp

```diff
@@ -279,6 +335,21 @@ class db_val
 		return state_block_w_sideband_v14;
 	}
 
+	explicit operator nano::state_block_w_sideband () const
+	{
+		nano::bufferstream stream (reinterpret_cast<uint8_t const *> (data ()), size ());
+		auto error (false);
+		nano::state_block_w_sideband state_block_w_sideband;
+		state_block_w_sideband.state_block = std::make_shared<nano::state_block> (error, stream);
+		assert (!error);
+
+		state_block_w_sideband.sideband.type = nano::block_type::state;
+		error = state_block_w_sideband.sideband.deserialize (stream);
+		assert (!error);
+
+		return state_block_w_sideband;
+	}
```

Comment:
As this is essentially the same as `state_block_w_sideband_v14` it could be a refactored into a function template and then used in both places.